script/tool/lib/src/pubspec_check_command.dart

```diff
@@ -553,6 +554,19 @@ class PubspecCheckCommand extends PackageLoopingCommand {
         }
       });
     }
+
+    // Ensure that dev-only dependencies aren't in `dependencies`.
+    const List<String> devOnlyDependencies = <String>['integration_test'];
+    // non published packages like pidgeon subpackages are allowed to violate
```

Comment:
Non-published

Reply:
fixed

Reply:
@stuartmorgan @reidbaker should this same lint also be applied to dependencies like `test` or `flutter_test` ?

Reply:
Probably; we'd need an audit of the repo (e.g., by adding them and running the script locally) to see if we have any legitimate cases for it. I'd be happy to review a PR.

Reply:
@stuartmorgan what is the command to run updated `pubspec_check_command.dart` on the `packages` repository? Also, what launch config to use to be able to debug this script?

Also, wouldn't this lint be useful as part of the `https://pub.dev/packages/flutter_lints` or `https://pub.dev/packages/lints` ?

Reply:
> @stuartmorgan what is the command to run updated `pubspec_check_command.dart` on the `packages` repository? Also, what launch config to use to be able to debug this script?

The tool is documented [here](https://github.com/flutter/packages/blob/main/script/tool/README.md), but I'm not sure what you mean by launch config.

> Also, wouldn't this lint be useful as part of the `https://pub.dev/packages/flutter_lints` or `https://pub.dev/packages/lints` ?

It's not an actual lint, it's custom repository tooling, so it can't be added to packages that are just lists of lints to enable.

Reply:
> > @stuartmorgan what is the command to run updated `pubspec_check_command.dart` on the `packages` repository? Also, what launch config to use to be able to debug this script?
> 
> The tool is documented [here](https://github.com/flutter/packages/blob/main/script/tool/README.md), but I'm not sure what you mean by launch config.

Okay, it looks like the CI runs it with these params:

`dart run script/tool/bin/flutter_plugin_tools.dart pubspec-check --allow-dependencies=script/configs/allowed_unpinned_deps.yaml --allow-pinned-dependencies=script/configs/allowed_pinned_deps.yaml --min-min-flutter-version=3.13.0`

Adding `test` and `flutter_test` to the check brings up these packages:
```
  packages/css_colors
  packages/web_benchmarks
```

> > Also, wouldn't this lint be useful as part of the `https://pub.dev/packages/flutter_lints` or `https://pub.dev/packages/lints` ?
> 
> It's not an actual lint, it's custom repository tooling, so it can't be added to packages that are just lists of lints to enable.

Understand. Though the `dart analyze` is also being run by the CI. E.g. the `sort_pub_dependencies` lint is enforced on pubspecs.

I'm just saying maybe such test dependency check could/should be moved into lints. I've seen a few published packages messing test dependencies and Dart pub tools don't allow to specify dependency excludes down stream.

Reply:
> Adding `test` and `flutter_test` to the check brings up these packages:
> 
> ```
>   packages/css_colors
>   packages/web_benchmarks
> ```

Those both look like mistakes that could be fixed in a PR adding the check.

> I'm just saying maybe such test dependency check could/should be moved into lints.

You could certainly propose new lints in the Dart linter repo. Many things in flutter/packages repo tooling are things that in theory I would like to explore moving to various upstream places (Dart lints, pub, etc.), but don't have the time currently.

Reply:
> > Adding `test` and `flutter_test` to the check brings up these packages:
> > ```
> >   packages/css_colors
> >   packages/web_benchmarks
> > ```

Okay. Let's try this https://github.com/flutter/packages/pull/6472